docs/contributing/mentorship.md

```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
+* Allocate some time upfront for this process, as you might have a lot of mentee applications to look through.
+  * If possible, book some time on your calendars and discuss the selection process with co-mentors in a particular quarter beforehand.
+* Try to have some prerequisites in mind while shortlisting an application of a potential mentee (CV/questions/proposal/interview),
+  * The mentee should be able to allocate enough time to the mentorship.
+    * Thus, having prior important commitments like a full-time job or large educational course load can lead to mentees not having enough time for the mentorship, even if they commit to working > 30hrs a week. This could manifest as mentees not having enough energy to think about a project, skipping meetings, and generally not being able to utilize the time mentors can spend with them.
+    * So it is better to practically judge the amount of work a mentee would be able to balance before selecting them.
+    * Generally we expect our mentees to be university/college students, who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution.
+  * The mentee should have at least one prior contribution to Thanos or its dependencies that affect it indirectly. The contribution should demonstrate effort & interest on part of the potential mentee.
+    * We want to understand that the potential mentee is interested in the project and has invested some time and effort in trying to understand Thanos.
+    * However, the contribution should not be a minimal effort, such as a typo/grammar fix or variable name change. While these are valued contributions, they cannot effectively demonstrate mentee effort, and should instead be some code/doc change. Note that this doesn't need to be in a merged/finalized state and can be open.
+  * The mentee should have proper means of communication (can be ensured during the 20-minute interview).
+    * This constitutes things like good enough internet to be able to video calls without dropping, as that is the primary means of communication between mentors and mentees.
+    * As most of the discussions (meetings/issues/reviews/code comments) are conducted in English, the mentee should be able to communicate well in English (first/second language).
+  * The mentee should have some form of local development environment setup, that would not require the mentor to constantly debug the mentee's environment, while working on the project.
+    * This is hard to ensure fully, but generally the mentee should be working on linux/linux VM/macOS, so most generic GNU/BSD commands are compatible.
+    * The mentee should have basic git, bash, programming, and Go knowledge. 
+    * They should be using some common code editor/IDE (VSCode/Goland), for the duration of mentorship, so that it is easier to pair program and share knowledge with them, without being blocked by things like code navigation.
```

Comment:
This is odd to me. I'm a maintainer and a mentor and I've never used vscode or goland and would not expect any mentee to be required to do so, especially when the tools (like goland) are closed-source and paid.

Reply:
So the main motivation for putting this here is to ensure that the mentee uses some editor/IDE that allows them to navigate code and lookup definitions easily across large codebases. We've had some experiences where mentees weren't progressing or getting frequently blocked due to using a very simplistic text editor or something so complex, that they don't know how to operate it well. So they get confused when trying to think of solutions, and have to ask code-level questions frequently. This can be avoided if mentors already know about this while interviewing, and can suggest they switch to something easier, to have a more productive mentorship!

That being said, let me rephrase this line to capture this, and not make it seem like vscode/goland are the only options! 🙂

Reply:
Have updated this!